src/deluge/processing/stem_export/stem_export.cpp

```diff
@@ -316,6 +329,42 @@ int32_t StemExport::exportInstrumentStems(StemExportType stemExportType) {
 	return totalNumOutputs;
 }
 
+/// iterates through all instruments, checking if there's any that should be exported (unmuted)
+/// then exports them all as a single arrangement stem
+/// simulates the button combo action of pressing record + play twice to enable resample
+/// and stop recording at the end of the arrangement
+int32_t StemExport::exportArrangementStem(StemExportType stemExportType) {
+	// prepare all the instruments for stem export
+	int32_t totalNumOutputs = disarmAllInstrumentsForStemExport();
```

Comment:
it would be nice to add master track as a new stem export type and skip this entirely. That would avoid the messy logic in the disarming function